core/trino-main/src/main/java/io/trino/sql/analyzer/Analyzer.java

```diff
@@ -69,8 +70,8 @@ public class Analyzer
         this.session = requireNonNull(session, "session is null");
         this.analyzerFactory = requireNonNull(analyzerFactory, "analyzerFactory is null");
         this.statementAnalyzerFactory = requireNonNull(statementAnalyzerFactory, "statementAnalyzerFactory is null");
-        this.parameters = parameters;
-        this.parameterLookup = parameterLookup;
+        this.parameters = ImmutableList.copyOf(requireNonNull(parameters, "parameters is null"));
+        this.parameterLookup = ImmutableMap.copyOf(requireNonNull(parameterLookup, "parameterLookup is null"));
```

Comment:
requireNonNull is unnecessary, since copyOf() will fail with an NPE if the argument is null.

Reply:
The error message is NPE not the readable such as `parameters is null`